### ISSUE TITLE
0xC0000005 when loading 30B model

### ISSUE THREAD

**Issue by HWiese1980** [oobabooga/text-generation-webui#1581]
### Describe the bug

I am trying to load MetalX/GPT4-X-Alpaca-30B-4bit but Oobabooga crashes with a 0xC0000005 error code. No additional message. 

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

Current commit b57ffc2e

1. Download MetalX/... model (as mentioned above)
2. Try loading it in Oobabooga text-generator-ui
3. Watch model load, until it crashes after "Loading model..." on the console

### Logs

```shell
No stack trace. Plain old crash.

I think it happens at `model.load_state_dict(safe_load(checkpoint), strict=False)` around line 73 in GPTQ_loader.py.
```

### System Info

```shell
Windows 11
32GB RAM
24GB VRAM (3090Ti)
```

**Comment by kuso-ge** [comment]
parameters?

**Comment by askmyteapot** [comment]
I have this issue randomly. With 32GB of RAM on windows, you need to manually set your swapfile to at least 140GB. And if your model is on a secondary drive, set a swap file for it too at like 40GB. Since i did that, i can load the large 30B files 9 times out of 10. 
On linux, i just set the upper 24GB of RAM to ZSWAP (compressed ram swap) and it loads no problems. 
If i was smarter, i would try to get KoboldAI's lazy loading working in Ooba, but alas... I has teh dum.

**Comment by HWiese1980** [comment]
Wonder why I would have to set an extra swap file for the drive the model is on... That doesn't make much sense to me. 

I'll see if I can make a swap file of that size. Or buy more RAM... I suppose though there are ways to reduce the required memory. For example, isn't there a way to load the model **in chunks** into the GPU? That would help a lot!

**Comment by HWiese1980** [comment]
Relevant parameters:

`--model gpt4-x-alpaca-30b-4bit --xformers --auto-devices --wbits 4 --groupsize 128 --model_type llama --gpu-memory 20`

**Comment by askmyteapot** [comment]
As for the 2nd swap file... i just found having that secondary one seemed to increase the likelihood of the model loading. 
I'm debating whether i get myself an extra 2x32GB ddr4 kit. 

And with your parameters, i would drop --auto-devices and --gpu-memory. As thats all set in the ui now. And if you are using your 3090ti for primary display you only will have ~22.5GB of available VRAM. Plus, 128 groupsize 30B models use more than 24GB at full context. You are better off getting a non-groupsize quantized model, as you will be able to run a bit higher context (full ctx if the card isnt used to display)

**Comment by HWiese1980** [comment]
Yeah, card is used for display. I would have expected to get a OOM error from CUDA if there's not enough VRAM. 

I'll try with your advice, see what that does... I remember I had it running once. It was slow, but it worked. So there has to be enough VRAM in total, at least under certain circumstances. 

I have 2x16GB DDR4-3600. I'm thinking of adding another 2x16 of the same type to that. Or do you think, 64GB are still too little?

**Comment by askmyteapot** [comment]
So this was my system idle before loading the model
![before ooba](https://user-images.githubusercontent.com/62238146/235130484-e175e7c8-dabb-4476-a876-2b00e14e0ad3.png)

And this is peak while loading.
![LOADING MODEL](https://user-images.githubusercontent.com/62238146/235130531-30a2cfa2-da96-4367-b023-a43e30e8df10.png)

**Comment by askmyteapot** [comment]
And as for the slowness, with the auto-devices and memory, it was probably loading some layers onto the CPU, which is why it didnt OOM, but was slow. 

For reference, on my Tesla P40 (think 1080ti), with low context, i get 4-6tokens a second. At full context, i get between 1-2t/s.
Your 3090to should be getting i believe 4-6t/s at near full context.

**Comment by askmyteapot** [comment]
So this is a 13B model at low context
![image](https://user-images.githubusercontent.com/62238146/235132321-45c8b308-eeea-4675-af2c-f7768fb36e80.png)
and high context

and this is an example of the speeds i get on the P40 as the context size increases. The low rates are when it only generates a limited number of tokens. 
![image](https://user-images.githubusercontent.com/62238146/235133962-d85b2875-0793-4866-afeb-615c8f18a83f.png)

**Comment by HWiese1980** [comment]
Yeah, just for everyone's info: buying another 32GB of system RAM has solved the issue for me. It seems like the cause was indeed too little memory. A proper error message would have been nice though.
